rtl/templates/sonata_pkg.sv.tpl

```diff
@@ -6,12 +6,22 @@
 
 package sonata_pkg;
 
+  // Number of Instances
   % for block in config.blocks:
   localparam int unsigned ${block.name.upper()}_NUM = ${block.instances};
   % endfor
 
+  // Width of block IO arrays
+  % for block in config.blocks:
+  % for b_io in block.ios:
+  % if b_io.length is not None:
+  localparam int unsigned ${f"{block.name}_{b_io.name}_WIDTH".upper()} = ${b_io.length};
+  % endif
+  % endfor
+  % endfor
```

Comment:
Looks a bit complicated for just producing the GPIO width.

Reply:
I don't think this is too complicated for what it is. It produces a width parameter for every block IO which has a length. So it can also be used for SPI CS.

I think this is a pretty clear way of expressing what it does

Reply:
Ok, it doesn't look wrong to me so let's keep it in.